src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Rejection.scala

```diff
@@ -29,23 +38,23 @@ object Rejection {
     *
     * @param ref a reference to the resource
     */
-  final case class IsDeprecated(ref: Ref) extends Rejection(s"Resource '${ref.show}' is deprecated.")
+  final case class ResourceIsDeprecated(ref: Ref) extends Rejection(s"Resource '${ref.show}' is deprecated.")
 
   /**
     * Signals an attempt to interact with a resource that is expected to be a file resource but it isn't.
     *
     * @param ref a reference to the resource
     */
-  final case class NotFileResource(ref: Ref) extends Rejection(s"Resource '${ref.show}' is not a file resource.")
+  final case class NotAFileResource(ref: Ref) extends Rejection(s"Resource '${ref.show}' is not a file resource.")
 
   /**
     * Signals an attempt to perform a request with an invalid payload.
     *
     * @param ref a reference to the resource
-    * @param reason the human readable reason for the rejection
+    * @param details the human readable reason for the rejection
     */
-  final case class InvalidPayload(ref: Ref, reason: String)
-      extends Rejection(s"Resource '${ref.show}' with invalid payload due to '$reason'.")
+  final case class InvalidPayload(ref: Ref, details: String)
```

Comment:
In admin we have something like `InvalidProjectFormat`. Maybe we could name it `InvalidResourceFormat` ?